src/appleseed/renderer/kernel/rendering/progressive/progressiveframerenderer.cpp

```diff
@@ -222,7 +231,8 @@ namespace
 
         bool is_rendering() const override
         {
-            return m_job_queue.has_scheduled_or_running_jobs();
+            m_stopwatch.measure();
+            return (m_job_queue.has_scheduled_or_running_jobs() && m_params.m_time_limit > m_stopwatch.get_seconds());
```

Comment:
We need to revert this: even if the maximum rendering time has been reached, if there are still running jobs, this method should return true. If it lies and returns false, it's going to cause bugs of all sorts.

The proper way to implement this is to use a `IRendererController`. The host application already passes one, maybe we need to wrap it into another one that implements time limiting.